crates/storage/provider/src/providers/blockchain_provider.rs

```diff
@@ -152,9 +152,16 @@ where
     }
 
     fn header_td_by_number(&self, number: BlockNumber) -> ProviderResult<Option<U256>> {
-        self.database.header_td_by_number(number)
+        if let Some(td) = self.database.header_td_by_number(number)? {
+            Ok(Some(td))
+        } else {
+            let last_persisted_block_number = self.database.last_block_number()?;
+            self.database.header_td_by_number(last_persisted_block_number)
```

Comment:
Assuming post-merge, this makes sense, because the td should be the same for the highest persisted block.

Let's add a comment here to explain why this works in PoS